File: segundos 4 lenguajes/Java/JugadorNormal.java
package Java;
public class JugadorNormal  {
    String nombre;
    int nivel;

    // Constructor sin parámetros
    public JugadorNormal()  {
        this.nombre = "Juan";
        this.nivel = 10;

        System.out.println("El jugadador normal es: " + nombre + ""+ nivel );
    }

    // Constructor con parámetros
    public JugadorNormal(String nombre, int nivel) {
        this.nombre = nombre;
        this.nivel = nivel;

    
    }

    // Constructor sobrecargado (solo nombre)
    public JugadorNormal(String nombre) {
        this.nombre = nombre;
        this.nivel = 1;
    }

    // Constructor copia
    public JugadorNormal(JugadorNormal copia) {
        this.nombre = copia.nombre;
        this.nivel = copia.nivel;

        System.out.println();
    }
}

File: segundos 4 lenguajes/go/JugadorNormal.go
package main

type JugadorNormal struct {
	Nombre string
	Nivel  int
}

func NewJugadorNormal() JugadorNormal {
	return JugadorNormal{"Juan", 10}
}

func NewJugadorNormalParams(nombre string, nivel int) JugadorNormal {
	return JugadorNormal{nombre, nivel}
}

func NewJugadorNormalNombre(nombre string) JugadorNormal {
	return JugadorNormal{nombre, 1}
}

func NewJugadorNormalCopy(j JugadorNormal) JugadorNormal {
	return JugadorNormal{j.Nombre, j.Nivel}
}
